daemon/stats.go

```diff
@@ -126,6 +129,20 @@ func (daemon *Daemon) ContainerStats(ctx context.Context, prefixOrName string, c
 			if !config.Stream {
 				return nil
 			}
+		case <-time.After(statsInterval):
+			// if no stream, we need to make sure to send real data
+			if !config.Stream {
```

Comment:
We can't.

If `--no-stream` is set, client will fetch the stat data once and only once then quit, if we return nil here, client will never get the data, so we need to `continue` to try again to send the real data.

Besides, I use `continue` here is also for interrupting next step of sending empty stats to client, but try one more again to send the real data. 

Integration test `TestStatsAllRunningNoStream` is for this use case.